packages/ember-data/lib/serializers/json-api-serializer.js

```diff
@@ -0,0 +1,299 @@
+/**
+  @module ember-data
+*/
+
+import JSONSerializer from 'ember-data/serializers/json-serializer';
+import normalizeModelName from 'ember-data/system/normalize-model-name';
+import { pluralize, singularize } from 'ember-inflector/lib/system/string';
+
+var dasherize = Ember.String.dasherize;
+
+/**
+  @class JSONAPISerializer
+  @namespace DS
+  @extends DS.JSONSerializer
+*/
+export default JSONSerializer.extend({
+
+  /**
+    @method _normalizeResponse
+    @param {DS.Store} store
+    @param {DS.Model} primaryModelClass
+    @param {Object} payload
+    @param {String|Number} id
+    @param {String} requestType
+    @param {Boolean} isSingle
+    @return {Object} JSON-API Document
+    @private
+  */
+  _normalizeResponse: function(store, primaryModelClass, payload, id, requestType, isSingle) {
+
+    let normalizeResourceHelper = (resourceHash) => {
+      let modelName = this.modelNameFromPayloadKey(resourceHash.type);
+      let modelClass = this.store.modelFor(modelName);
+      let serializer = this.store.serializerFor(modelName);
+      let { data } = serializer.normalize(modelClass, resourceHash);
+      return data;
+    };
+
+    if (Ember.typeOf(payload.data) === 'object') {
+      payload.data = normalizeResourceHelper(payload.data);
+    } else {
+      payload.data = payload.data.map(normalizeResourceHelper);
+    }
+
+    if (Ember.typeOf(payload.included) === 'array') {
+      payload.included = payload.included.map(normalizeResourceHelper);
```

Comment:
need to use Ember.EnumerableUtils if it's available, otherwise this will break IE8